.travis.yml

```diff
@@ -1,76 +1,100 @@
-
-
-dist: xenial
 sudo: required
 language:
   - generic
+compiler:
+  - gcc
 cache:
   - apt
+services:
+  - docker
 
 env:
   global:
     - ROS_DISTRO=kinetic
     - ROS_CI_DESKTOP="`lsb_release -cs`"  # e.g. [precise|trusty|...]
-    - CI_SOURCE_PATH=$(pwd)
-    - ROSINSTALL_FILE=$CI_SOURCE_PATH/dependencies.rosinstall
-    - CATKIN_OPTIONS=$CI_SOURCE_PATH/catkin.options
     - ROS_PARALLEL_JOBS='-j8 -l6'
-    # Set the python path manually to include /usr/-/python2.7/dist-packages
-    # as this is where apt-get installs python packages.
     - PYTHONPATH=$PYTHONPATH:/usr/lib/python2.7/dist-packages:/usr/local/lib/python2.7/dist-packages
+    - BRANCH=$(if [ "$TRAVIS_PULL_REQUEST" == "false" ]; then echo $TRAVIS_BRANCH; else echo $TRAVIS_PULL_REQUEST_BRANCH; fi)
+
 
 ################################################################################
 
-# Install system dependencies, namely a very barebones ROS setup.
 before_install:
-  - sudo rm /var/lib/dpkg/lock
-  - sudo sh -c "echo \"deb http://packages.ros.org/ros/ubuntu $ROS_CI_DESKTOP main\" > /etc/apt/sources.list.d/ros-latest.list"
-  - sudo apt-key adv --keyserver hkp://ha.pool.sks-keyservers.net:80 --recv-key 421C365BD9FF1F717815A3895523BAEEB01FA116
-  - sudo apt-get update -qq
-  - sudo apt-get install -y python-catkin-pkg python-rosdep python-wstool ros-$ROS_DISTRO-catkin python-catkin-tools
-  - source /opt/ros/$ROS_DISTRO/setup.bash
-  # Prepare rosdep to install dependencies.
-  - sudo rosdep init
-  - rosdep update
+  - docker run -e BRANCH -e PYTHONPATH -d --name alica-test -v $TRAVIS_BUILD_DIR:/travis ros:kinetic-ros-core-xenial tail -f /dev/null
```

Comment:
why are you tailing /dev/null?

Reply:
If docker run is not provided a command then the container shuts down. tailing /dev/null allows the container to stay open so I can pass it commands later as I'm doing in the several calls to exec.